diff --git i/services/arm_arch_svc/arm_arch_svc_setup.c w/services/arm_arch_svc/arm_arch_svc_setup.c
index 469af471a..07ba33c9c 100644
--- i/services/arm_arch_svc/arm_arch_svc_setup.c
+++ w/services/arm_arch_svc/arm_arch_svc_setup.c
@@ -51,6 +51,7 @@ static int32_t smccc_arch_features(u_register_t arg1)
 	case SMCCC_ARCH_WORKAROUND_2: {
 #if WORKAROUND_CVE_2018_3639
 #if DYNAMIC_WORKAROUND_CVE_2018_3639
+#if 0
 		/*
 		 * Firmware doesn't have to carry out dynamic workaround if the
 		 * PE implements architectural Speculation Store Bypass Safe
@@ -60,6 +61,7 @@ static int32_t smccc_arch_features(u_register_t arg1)
 			return SMC_WA_DO_NOT;
 		}
 
+#endif
 		switch (check_erratum_applies(ERRATUM(ARCH_WORKAROUND_2))) {
 		case ERRATA_APPLIES:
 			return SMC_WA_DO;
